lib/src/extension/register_extension.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:injector/injector.dart';
+import 'package:injector/src/factory/factory.dart';
+
+extension RegisterExtension on Injector {
+  void registerSingleton<T>(
+    Builder<T> builder, {
+    bool override = false,
+    String dependencyName = "",
+  }) =>
+      this.register(Factory.singleton(builder), override: override, dependencyName: dependencyName);
+
+  void registerDependency<T>(
+    Builder<T> builder, {
+    bool override = false,
+    String dependencyName = "",
+  }) =>
+      this.register(Factory.singleton(builder), override: override, dependencyName: dependencyName);
```

Comment:
should be ```Factory.factory ```. 

Let's write some tests. this could be a dangerous change

Reply:
Nice catch! I also named the factory `provider` instead of `factory` because it was weird to have `Factory.factory`. I was looking for a more descriptive way to show that you create the instance overy time again 🤔